File: Exercises/RB101-RB119_Small_Problems/Easy_8/ex9.rb
# frozen_string_literal: true

# Write a method that takes one argument, a string containing one or more words,
# and returns the given string with words that contain five or more characters
# reversed. Each string will consist of only letters and spaces. Spaces should
# be included only when more than one word is present.

def reverse_words(str)
  str.split.map do |word|
    if word.size > 4
      word.chars.reverse.join
    else
      word
    end
  end.join(' ').strip
end

puts reverse_words('Professional') == "lanoisseforP"          # => lanoisseforP
puts reverse_words('Walk around the block') == "Walk dnuora the kcolb" # => Walk dnuora the kcolb
puts reverse_words('Launch School') == "hcnuaL loohcS"         # => hcnuaL loohcS

File: Exercises/JavaScript/JavaScript_Basics/Strings/ex9.js
// Write code that capitalizes the words in the string 'launch school tech & talk', so that you get the string 'Launch School Tech & Talk'.

let str = 'launch school tech & talk';
let str2 = str.split(' ');
for( let i = 0; i < str2.length; i += 1 ) {
  str2[i] = str2[i][0].toUpperCase() + str2[i].slice(1);
}
console.log(str2.join(' '));
